static/scripts/secret.ts

```diff
@@ -325,6 +327,12 @@ const setHandler = async () => {
         const parsedConf: IConf | undefined = await parseYAML(conf);
         const advParsed: IConf | undefined = parsedAdv;
         updatedConf[KEY_NAME] = encryptedValue;
+        updatedConf["chain-id"] =
+          advParsed && advParsed["chain-id"] && !Number.isNaN(Number(advParsed["chain-id"]))
+            ? Number(advParsed["chain-id"])
+            : parsedConf && parsedConf["chain-id"] && !Number.isNaN(Number(parsedConf["chain-id"]))
+            ? Number(parsedConf["chain-id"])
+            : Number(defaultConf["chain-id"]);
```

Comment:
Can you rewrite this as a function to increase readability? The others should also be rewritten as well.

Reply:
refactored